test/WebJobs.Script.Tests/WorkerFunctionMetadataProviderTests.cs

```diff
@@ -221,5 +222,74 @@ public void ValidateFunctionMetadata_IsoStringNotAltered()
+            var mockChannelManager = new Mock<IWebHostRpcWorkerChannelManager>(MockBehavior.Strict);
+            var mockScriptHostManager = new Mock<IScriptHostManager>(MockBehavior.Strict);
+            var mockOptionsMonitor = new Mock<IOptionsMonitor<ScriptApplicationHostOptions>>(MockBehavior.Strict);
+            var scriptOptions = new ScriptApplicationHostOptions
+            {
+                IsFileSystemReadOnly = true
+            };
+            mockOptionsMonitor.Setup(m => m.CurrentValue).Returns(scriptOptions);
+
+            var testEnvironment = new TestEnvironment();
+            testEnvironment.SetEnvironmentVariable("FUNCTIONS_WORKER_RUNTIME", "node");
+
+            var mockRpcWorkerChannel = new Mock<IRpcWorkerChannel>(MockBehavior.Strict);
+            var rawFunctionMetadataList = new List<RawFunctionMetadata>
+            {
+                new RawFunctionMetadata
+                {
+                    Metadata = new FunctionMetadata { Name = "TestFunction" },
+                    Bindings = ["{\"type\": \"httpTrigger\", \"name\": \"req\", \"direction\": \"in\"}"],
+                    UseDefaultMetadataIndexing = false
+                }
+            };
+            mockRpcWorkerChannel.Setup(m => m.GetFunctionMetadata()).ReturnsAsync(rawFunctionMetadataList);
+
+            var tcs = new TaskCompletionSource<IRpcWorkerChannel>();
+            tcs.SetResult(mockRpcWorkerChannel.Object);
+            var channels = new Dictionary<string, TaskCompletionSource<IRpcWorkerChannel>>
+            {
+                { "testWorkerId", tcs }
+            };
+
+            mockChannelManager.Setup(m => m.GetChannels("node")).Returns(channels);
+
+            var provider = new WorkerFunctionMetadataProvider(
+                mockOptionsMonitor.Object,
+                NullLogger<WorkerFunctionMetadataProvider>.Instance,
+                testEnvironment,
+                mockChannelManager.Object,
+                mockScriptHostManager.Object);
+
+            var workerConfigs = new List<RpcWorkerConfig>();
+
+            // Calling this twice should return the same data
+            var result1 = await provider.GetFunctionMetadataAsync(workerConfigs, true);
+            var result2 = await provider.GetFunctionMetadataAsync(workerConfigs, true);
+
+            var function1 = result1.Functions.Single();
+            var function2 = result2.Functions.Single();
+
+            static void AssertFunction(FunctionMetadata function)
+            {
+                Assert.Equal("TestFunction", function.Name);
+                Assert.Equal("node", function.Language);
+                Assert.Collection(function.Bindings, binding =>
+                {
+                    Assert.Equal("httpTrigger", binding.Type);
+                    Assert.Equal("req", binding.Name);
+                    Assert.Equal(BindingDirection.In, binding.Direction);
+                });
+            }
+
+            AssertFunction(function1);
```

Comment:
this test failed before the fix... there would have been multiple "httpTrigger" bindings, but we now check that there's only one